woocommerce/woocommerce-android
# Shipping Label Packages: Step 2

The second part of #3019, this adds:

1. The packages selector screen, and handling its interactions/result (b17862a8e9c7fe78ffdc422bd298b96274fce76b, 4e9d924207da9cb7ac60e9a033152680a2b29881, 485ec0031d28f1d234b8deb1ed92aebd6d2aba08).
2. Handle data validation in the package details screen (4035abf80f8b00c4232982859e720307f743037d).
3. Update the state machine to handle package details selection (81a539f98938e8e7282904fb03e5e5d6bcdd1c52).
4. Make the `selectedPackage` nullable, and stop using the first package in the list by default (b9e322db6e3bd261817b8bee13ae772f87646a5c), I'll update FluxC to handle the call to `/wc/v1/connect/account/settings` to get the last used package.
5. Update error handling if fetching products fail (46928aff10181e26ed79b3050de16976d5a62d99, p1612193926177800/1611766213.104900-slack-C6H8C3G23)

The rest of the task + unit tests will be handled in upcoming PR.

<img src="https://user-images.githubusercontent.com/1657201/106601208-af281100-655b-11eb-805e-6ee19e18172d.gif" width=400 />

#### Testing

1. Prepare an order on the web that satisfies SL creation preconditions.
2. Go to Orders -> Open the order
3. Pass the steps of origin and destination addresses.
4. Click on Continue of Package details.
5. Notice that a loading dialog appears, and then the package details screen will be displayed.
6. Click on the "selected package" spinner.
7. Notice the packages selection screen is opened.
8. Select a package.
9. Notice the package details screen is updated with the selected package.
10. Enter a weight > 0.
11. Click on "done" button.
12. Notice the packages details step is completed.
13. Click on "edit" in the package details step.
14. Notice the package details screen contains the last changes.

Update release notes:
